### PR TITLE
Add more info to Contributing.md

### PR DESCRIPTION
**Pull request by malpunek**
Added info for potential developers

---
This change is [Reviewable](https://reviewable.io/reviews/codilime/veles/399)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+
+#### Contributing pipeline
+
+* Create your own branch. Name it `wip/your github nick/descriptive-branch-name`
+* Develop your branch. Remember to:
+  * keep your commit atomic (one feature per commit)
+  * add descriptive commit messages. One-liners are fine for small, obvious changes, but bigger changes should have longer commit messages.
+  * make sure email in your commits metadata is linked with your github account
+  * Use `format` build target (for Linux run `make format`) relatively often to avoid huge commits "Style fix"
+* When you think you're done open Github PR with a clear list of what you've done. From now on your branch is `public` and you cannot forcepush (`git push -f`) anything to it. If you really need to/want to pushforce FIRST ask reviewers for permission
```
[the remark this answers is not in the document]
Comment (mkow) [reply CONTRIBUTING.md]: `permission` -> `permission.`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+* Every PR has to pass clang-tidy checks (`lint` build target).
+* Submit fixes according to the review in new commits
```
Comment (mkow) [inline CONTRIBUTING.md]: end the sentence with a dot.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+* When your code will be accepted, please squash your fixes into previous feature commits
+* Your code will be merged into master
 
+More information about pull requests you can find [here](https://guides.github.com/activities/contributing-to-open-source/#contributing).
```
Comment (mkow) [inline CONTRIBUTING.md]: Should be rather:
> You can find more information about pull requests [here]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+* Create your own branch. Name it `wip/your github nick/descriptive-branch-name`
+* Develop your branch. Remember to:
+  * keep your commit atomic (one feature per commit)
+  * add descriptive commit messages. One-liners are fine for small, obvious changes, but bigger changes should have longer commit messages.
+  * make sure email in your commits metadata is linked with your github account
+  * Use `format` build target (for Linux run `make format`) relatively often to avoid huge commits "Style fix"
+* When you think you're done open Github PR with a clear list of what you've done. From now on your branch is `public` and you cannot forcepush (`git push -f`) anything to it. If you really need to/want to pushforce FIRST ask reviewers for permission
+* Every PR has to pass clang-tidy checks (`lint` build target).
+* Submit fixes according to the review in new commits
+* When your code will be accepted, please squash your fixes into previous feature commits
```
Comment (mkow) [inline CONTRIBUTING.md]: I'm not sure whether this is grammatically correct

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+* Your code will be merged into master
 
+More information about pull requests you can find [here](https://guides.github.com/activities/contributing-to-open-source/#contributing).
+
+#### File placement
+We have some rules as to where to place new files:
+* All things that depend on QT should be placed in *ui* folder
```
Comment (mkow) [inline CONTRIBUTING.md]: `QT` -> `Qt`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -13,25 +31,45 @@ reported. Gather following information:
...
+* There should be as few things in *util* folder as possible. *util* is only for pieces of code that don't belong to any existing folders and are to small to have their own folder
+
+<br><br>
 Thanks for taking your time to contribute to Veles!
+
+*Powered by*<br>
```
Comment (mkow) [inline CONTRIBUTING.md]: no need for this huge image :P

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,3 +1,21 @@
+[//]: # (Copyright 2018 CodiLime.
```
Comment (mkow) [inline CONTRIBUTING.md]: IMO no need for a license here, we don't have anything important here.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
-clang-tidy checks (`lint` target). [BUILDING.rst](BUILDING.rst) contains a
-short guide describing how to configure them.
 
-Always write a clear log message for your commits describing the change.
-One-liners are fine for small, obvious changes, but bigger changes should have
-longer commit messages.
-
-More information about pull requests you can find
-[here](https://guides.github.com/activities/contributing-to-open-source/#contributing).
+#### Formal requirement
```
Comment (mkow) [inline CONTRIBUTING.md]: Why not three hashes? Previous level has two of them.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
 You **NEED** to have a signed CLA sent to CodiLime before your code can be
 merged. You can download one from [Contributor License
 Agreement](https://www.codisec.com/cla). Sign it and then send a scan of it to
 contact@codisec.com. You can open a pull request and have it reviewed while we
 process your CLA.
+ 
+#### Style
+We are trying to stick as close as possible to [Google C++ Style Guide].
+Refer to this document whenever in doubt.
+With a few exceptions:
```
Comment (mkow) [inline CONTRIBUTING.md]: Remove `With`, it doesn't connect well with the previous sentence.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+* We use `#pragma once`
```
Comment (mkow) [inline CONTRIBUTING.md]: Use dots at the end of sentences (here and in other places).

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+* We have to use raw pointers in many places, because of Qt
+* We don't like the Hungarian notation used for constants, so maybe we will
+switch to CAPS_NOTATION at some point.
+* You might fing some older code that doesn't conform GCS
```
Comment (mkow) [inline CONTRIBUTING.md]: `fing`? ;)

Comment (mkow) [reply CONTRIBUTING.md]: it should be `conform to`

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+
+#### Contribution pipeline
+
+* Create your own branch. Name it `wip/your github nick/descriptive-branch-name`.
```
Comment (mkow) [inline CONTRIBUTING.md]: github -> `GitHub`

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+* Develop your branch. Remember to:
+  * Keep your commits atomic (one feature per commit).
+  * Add descriptive commit messages. One-liners are fine for small,
+   obvious changes, but bigger changes should have longer commit messages.
```
Comment (mkow) [inline CONTRIBUTING.md]: Please fix indent (one more space), also in other places.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+  * Make sure email in your commits metadata is linked with your GitHub
```
Comment (mkow) [inline CONTRIBUTING.md]: It'd rather write `the email`, but I'm not 100% sure.

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+   account.
+  * Use `format` build target (for Linux run `make format`) relatively
+   often to avoid huge "Style fix" commits.
+* When you think you're done open GitHub PR with a clear list of what you've
```
Comment (mkow) [inline CONTRIBUTING.md]: `a GitHub PR`

Comment (mkow) [reply CONTRIBUTING.md]: I'd also write `what you did`, but it's not an expert, so do what you prefer.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+done. From now on your branch is `public` and you shouldn't forcepush
```
Comment (mkow) [inline CONTRIBUTING.md]: Use emphasis instead of backticks (`*public*` in Markdown).

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+* Submit fixes according to the review in new commits.
+* Once your code is accepted by reviewers, please squash your fixes
+ into previous feature commits.
+* Your code will be merged into master.
 
+You can find more information about pull requests [here][pr_guide].
+
+#### File placement
+We have some rules as to where to place new files:
+* All things that depend on Qt should be placed in *ui* folder
```
Comment (mkow) [inline CONTRIBUTING.md]: Use backticks instead of emphasis for `ui`.

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+* There should be as few things in *util* folder as possible. *util* is only
```
Comment (mkow) [inline CONTRIBUTING.md]: ditto

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+for pieces of code that don't belong to any existing folders and are to small
```
Comment (mkow) [inline CONTRIBUTING.md]: `to small` -> `too small`

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+to have their own folder
+
+<br><br>
 Thanks for taking your time to contribute to Veles!
+
+   [Google C++ Style Guide]: <https://google.github.io/styleguide/cppguide.html>
```
Comment (mkow) [inline CONTRIBUTING.md]: No indentation here.

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -7,31 +7,64 @@ reported. Gather following information:
...
+   [ms_guide]: <https://support.microsoft.com/en-us/help/13443/windows-which-operating-system>
+   [pr_guide]: <https://guides.github.com/activities/contributing-to-open-source/#contributing>
```
Comment (mkow) [inline CONTRIBUTING.md]: Seems the link is outdated, please fix.

Comment (malpunek) [reply CONTRIBUTING.md]: the old link redirects here:
https://opensource.guide/how-to-contribute/

The new  page is less than helpful in terms of PRs. I would either delete the link (and assume people willing to make PRs know the basics about them) or point to a different page, for instance 
http://codeinthehole.com/tips/pull-requests-and-other-good-practices-for-teams-using-github/

Comment (mkow) [reply CONTRIBUTING.md]: just remove it, it's not necessary

**Comment by malpunek** [comment]
Review status: 0 of 1 files reviewed at latest revision, 25 unresolved discussions.

---

*[CONTRIBUTING.md, line 1 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-1:-L47017GQjk9NAasuBiv:b32w0x) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L1)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

IMO no need for a license here, we don't have anything important here.
</blockquote></details>

OK

---

*[CONTRIBUTING.md, line 44 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-44:-L470K07NbRLnzt_arba:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L44)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

With a few exceptions:
* We use `#pragma once`
* We have to use raw pointers in many places, because of Qt
* We don't like the Hungarian notation used for constants, so maybe we will switch to CAPS_NOTATION at some point.
* Some older code doesn't conform GCS
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 46 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-46:-L470iSa9hWPkI4qAY-T:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L46)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Hmm, not `Contribution pipeline`?
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 50 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-50:-L470jo7vKGpGngwOk7B:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L50)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`commit` -> `commits`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 53 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-53:-L4715dY9jGiMM6OnsUW:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L53)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`huge commits "Style fix"` -> `huge "Style fix" commits`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 54 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-54:-L471D_p5PeCqPqgqx31:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L54)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`permission` -> `permission.`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 56 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-56:-L471Gl_zixCx9B9DqDp:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L56)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

end the sentence with a dot.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 57 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-57:-L471HzbQwt7MkNOF2OQ:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L57)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

I'm not sure whether this is grammatically correct
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 60 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-60:-L471JdZEf3thmKLhs_t:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L60)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Should be rather:
> You can find more information about pull requests [here]
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 64 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-64:-L471KzdF9eW_hpNjChL:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L64)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`QT` -> `Qt`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 70 at r1](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r1-70:-L471NakGe16yCxtxXmA:b-896fix) ([raw file](https://github.com/codilime/veles/blob/39dc367581c54536c7d71a1dc6f6843a028328ee/CONTRIBUTING.md#L70)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

no need for this huge image :P
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 17 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-17:-L470IDrRiwEa0OvhxC7:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L17)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why not three hashes? Previous level has two of them.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 28 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-28:-L470ynXroHefqUpqcnD:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L28)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Remove `With`, it doesn't connect well with the previous sentence.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 29 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-29:-L470ckLMhwPEYq195gp:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L29)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Use dots at the end of sentences (here and in other places).
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 33 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-33:-L470hX-t28wyCbf0C73:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L33)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

it should be `conform to`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 37 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-37:-L470j4El2ANCo5jvKTq:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L37)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

github -> `GitHub`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 41 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-41:-L470l1BAuDdM-DbVkSs:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L41)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Please fix indent (one more space), also in other places.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 46 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-46:-L471FVD3xCjZG3jGow-:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L46)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

I'd also write `what you did`, but it's not an expert, so do what you prefer.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 47 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-47:-L471G7aBBSz1urvVaa6:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L47)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Use emphasis instead of backticks (`*public*` in Markdown).
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 60 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-60:-L471Lpk3Wnbv-mCMgL2:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L60)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Use backticks instead of emphasis for `ui`.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 61 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-61:-L471MIEqw6J1Dk3KIje:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L61)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 62 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-62:-L471N1GfdtvVxf6Pzdm:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L62)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`to small` -> `too small`
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 68 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-68:-L471O7_y60msGCrwbsr:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L68)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

No indentation here.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 70 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-70:-L471PDaHQgezbZKbJPi:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L70)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

just remove it, it's not necessary
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/codilime/veles/399)*

**Comment by malpunek** [comment]
Review status: 0 of 1 files reviewed at latest revision, 25 unresolved discussions.

---

*[CONTRIBUTING.md, line 42 at r2](https://reviewable.io:443/reviews/codilime/veles/399#-L47-Ev2ny6OFQraGHNR-r2-42:-L472JNEWlx7idWYJFIS:b-896fix) ([raw file](https://github.com/codilime/veles/blob/780af0cb52c8efe8391b6d080a42c7a4bf229f57/CONTRIBUTING.md#L42)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

It'd rather write `the email`, but I'm not 100% sure.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/codilime/veles/399)*

**Comment by mkow** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[CONTRIBUTING.md, line 48 at r5](https://reviewable.io:443/reviews/codilime/veles/399#-L5DwMYmWXsBTyiASlAI:-L5DwMYmWXsBTyiASlAJ:bo5u94j) ([raw file](https://github.com/codilime/veles/blob/d9e3232977aaeccd51f626ca6ad676364fefba0a/CONTRIBUTING.md#L48)):*
> ```Markdown
>     often to avoid huge "Style fix" commits.
> * When you think you're done open a GitHub PR with a clear list of what you
> did. From now on your branch is *public* and you shouldn't forcepush
> ```

Please indent all the wrapped lines (here and in other places also).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/codilime/veles/399)*

**Comment by malpunek** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[CONTRIBUTING.md, line 48 at r5](https://reviewable.io:443/reviews/codilime/veles/399#-L5DwMYmWXsBTyiASlAI:-L6QVgpWa-clXkZcpbC1:b-896fix) ([raw file](https://github.com/codilime/veles/blob/d9e3232977aaeccd51f626ca6ad676364fefba0a/CONTRIBUTING.md#L48)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Please indent all the wrapped lines (here and in other places also).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/codilime/veles/399)*

**Comment by chivay** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 65 at r6](https://reviewable.io:443/reviews/codilime/veles/399#-L6QoP2iEpOl61I08wz_:-L6QoP2iEpOl61I08wza:b-8306ad) ([raw file](https://github.com/codilime/veles/blob/7d93629f00ffda483c9ae91efbf8f4531d78a4b7/CONTRIBUTING.md#L65)):*
> ```Markdown
> 
> <br><br>
> Thanks for taking your time to contribute to Veles!
> ```

AFAIK "taking your time" == "do things slowly".  It's not really the message we're trying to convey. :wink:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/codilime/veles/399)*

**Comment by mkow** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[CONTRIBUTING.md, line 3 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9FaJnB1cdBi3LGmkoz:-L9FaJnB1cdBi3LGmkp-:bcw5pji) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L3)):*
> ```Markdown
> # Contributing to Veles
> ## Reporting bugs
> If you found a bug in Veles please check first if the problem has been already
> ```

`found` -> `find` ?

---

*[CONTRIBUTING.md, line 28 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9Fau-CBu5RIMWDAIXj:-L9Fau-CBu5RIMWDAIXk:bjdvr4h) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L28)):*
> ```Markdown
> We are trying to stick as close as possible to [Google C++ Style Guide].
> Refer to this document whenever in doubt.
> Usually:
> ```

Hmm, this is a list of exceptions to the guide, `usually` sounds strange to me here.

---

*[CONTRIBUTING.md, line 38 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9Fd8YDEMUl-RT6a_Nh:-L9Fd8YDEMUl-RT6a_Ni:b20mru6) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L38)):*
> ```Markdown
> ### Contribution pipeline
> 
> * Create your own branch. Name it `wip/your GitHub nick/descriptive-branch-name`.
> ```

`your GitHub nick` -> `your_GitHub_nick`

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/codilime/veles/399)*

**Comment by malpunek** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[CONTRIBUTING.md, line 3 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9FaJnB1cdBi3LGmkoz:-L9Fh1H_9wN1BTs3If6N:b-896fix) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L3)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`found` -> `find` ?
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 28 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9Fau-CBu5RIMWDAIXj:-L9Fh1-UEdg9xwuz4Irz:b-896fix) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L28)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Hmm, this is a list of exceptions to the guide, `usually` sounds strange to me here.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 38 at r7](https://beta.reviewable.io/reviews/codilime/veles/399#-L9Fd8YDEMUl-RT6a_Nh:-L9Fh-UW7cZDRhmYnFpW:b-896fix) ([raw file](https://github.com/codilime/veles/blob/0399ae015c0f7d226a6c046114aabb0011b7aa67/CONTRIBUTING.md#L38)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

`your GitHub nick` -> `your_GitHub_nick`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/codilime/veles/399)*

**Comment by mkow** [comment]
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/codilime/veles/399)*

**Pull request merged**